firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -218,45 +214,45 @@ public InstallationTokenResult generateAuthToken(
       @NonNull String fid,
       @NonNull String projectID,
       @NonNull String refreshToken)
-      throws FirebaseInstallationServiceException {
+      throws FirebaseException {
     String resourceName =
         String.format(GENERATE_AUTH_TOKEN_REQUEST_RESOURCE_NAME_FORMAT, projectID, fid);
     try {
-      URL url =
-          new URL(
-              String.format(
-                  "https://%s/%s/%s?key=%s",
-                  FIREBASE_INSTALLATIONS_API_DOMAIN,
-                  FIREBASE_INSTALLATIONS_API_VERSION,
-                  resourceName,
-                  apiKey));
+      int retryCount = 0;
+      while (retryCount <= MAX_RETRIES) {
+        URL url =
+            new URL(
+                String.format(
+                    "https://%s/%s/%s?key=%s",
+                    FIREBASE_INSTALLATIONS_API_DOMAIN,
+                    FIREBASE_INSTALLATIONS_API_VERSION,
+                    resourceName,
+                    apiKey));
 
-      HttpsURLConnection httpsURLConnection = (HttpsURLConnection) url.openConnection();
-      httpsURLConnection.setConnectTimeout(NETWORK_TIMEOUT_MILLIS);
-      httpsURLConnection.setReadTimeout(NETWORK_TIMEOUT_MILLIS);
-      httpsURLConnection.setDoOutput(true);
-      httpsURLConnection.setRequestMethod("POST");
-      httpsURLConnection.addRequestProperty("Authorization", "FIS_v2 " + refreshToken);
-      httpsURLConnection.addRequestProperty(CONTENT_TYPE_HEADER_KEY, JSON_CONTENT_TYPE);
-      httpsURLConnection.addRequestProperty(ACCEPT_HEADER_KEY, JSON_CONTENT_TYPE);
-      httpsURLConnection.addRequestProperty(CONTENT_ENCODING_HEADER_KEY, GZIP_CONTENT_ENCODING);
+        HttpsURLConnection httpsURLConnection = (HttpsURLConnection) url.openConnection();
+        httpsURLConnection.setConnectTimeout(NETWORK_TIMEOUT_MILLIS);
+        httpsURLConnection.setReadTimeout(NETWORK_TIMEOUT_MILLIS);
+        httpsURLConnection.setDoOutput(true);
```

Comment:
I don't see where we're writing a request (output) body. Do we still need this setDoOutput call?